compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3573,11 +3573,18 @@ private boolean equalityIntersectionExists(Set<BType> lhsTypes, Set<BType> rhsTy
             return true;
         }
 
-        boolean matchFound = lhsTypes
-                .stream()
-                .anyMatch(s -> rhsTypes
-                        .stream()
-                        .anyMatch(t -> isSameType(s, t)));
+        boolean matchFound = false;
+        for (BType lhsType : lhsTypes) {
+            for (BType rhsType : rhsTypes) {
+                if (isSameType(lhsType, rhsType) || isAssignable(lhsType, rhsType) || isAssignable(rhsType, lhsType)) {
+                    matchFound = true;
+                    break;
```

Comment:
can avoid the breaks with anymatch()

Reply:
Replaced the stream because it was suggested sometime ago that streams have a perf impact when it comes to trivial checks.